### PR TITLE
pt: refactor loss

### PR DESCRIPTION
**Pull request by iProzd**
This PR updates the loss interface to allow for a more flexible design. 
It enables processing input tensors before feeding them into the model, such as denoising operations (fyi @Chengqian-Zhang ). Previously, this was done in the data loader, which was less intuitive and more confusing. Now, users can easily handle these tasks within the loss function itself, as demonstrated in similar implementations in uni-mol: https://github.com/dptech-corp/Uni-Mol/blob/main/unimol/unimol/losses/unimol.py#L20.

### CODE REVIEW

**Comment by njzjz** [comment]
Could you make an explanation why it needs to change the interface?

**Comment by iProzd** [comment]
> Could you make an explanation why it needs to change the interface?

I've updated the comment.

File: deepmd/pt/loss/ener.py [hunk 1]
```diff
@@ -90,20 +90,26 @@ def __init__(
...
+        natoms : int
+            The local atom number.
 
         Returns
         -------
-        - loss: Loss to minimize.
+        loss: torch.Tensor
+            Loss for model to minimize.
+        more_loss: dict[str, torch.Tensor]
+            Other losses for display.
```
Comment (njzjz) [inline deepmd/pt/loss/ener.py:110]: The return values need to be updated.

File: deepmd/pt/loss/ener.py [hunk 2]
```diff
@@ -90,20 +90,26 @@ def __init__(
...
+    def forward(self, input_dict, model, label, natoms, learning_rate, mae=False):
+        """Return loss on energy and force.
 
-        Args:
-        - natoms: Tell atom count.
-        - p_energy: Predicted energy of all atoms.
-        - p_force: Predicted force per atom.
-        - l_energy: Actual energy of all atoms.
-        - l_force: Actual force per atom.
+        input_dict: dict[str, torch.Tensor]
```
Comment (njzjz) [inline deepmd/pt/loss/ener.py:96]:
```suggestion
        Parameters
        ----------
        input_dict: dict[str, torch.Tensor]
```

**Review by njzjz — APPROVED** [review]

**Review by Chengqian-Zhang — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]
